jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/StatsDWriter2.java

```diff
@@ -63,12 +66,15 @@ public StatsDWriter2(
 			@Nonnull String rootPrefix,
 			@Nonnull String bucketType,
 			boolean stringsValuesAsKey,
-			@Nonnull Long stringValueDefaultCount) {
+			@Nonnull Long stringValueDefaultCount,
+			@Nonnull String invalidCharValue) {
```

Comment:
You should pass directly the pattern here, and initialize it in the related Factory.